Action Extension/ActionExtensionViewController.swift

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2022 Metabolist. All rights reserved.
+
+import MobileCoreServices
+import UIKit
+import UniformTypeIdentifiers
+
+/// Let the user open a URL that might be a Mastodon profile or post in Metatext.
+/// Should activate for one or more web URLs, but will only do anything with the first..
+class ActionExtensionViewController: UIViewController {
+    /// Extensions aren't allowed to call `UIApplication.shared
+    ///  and thus don't have direct access to its `openURL` method.
+    /// `self.extensionContext?.open(<#T##URL: URL##URL#>)` only works for Today extensions.
+    /// As a workaround, we find a parent responder that has an `openURL(_:)` method.
+    /// This will be `UIApplication`. It's cursed, but it uses public APIs and works.
+    /// See <https://liman.io/blog/open-url-share-extension-swiftui>.
+    private func open(url: URL) {
+        var responder: UIResponder? = self as UIResponder
+        let selector = #selector(openURL(_:))
+        while responder != nil {
+            if responder!.responds(to: selector) && responder != self {
+                responder!.perform(selector, with: url)
+                return
+            }
+            responder = responder?.next
+        }
+    }
+
+    /// Only exists so we can create a selector from it.
+    @objc private func openURL(_ url: URL) {
+        return
+    }
+
+    /// This extension has no actual UI, so we act on extension input items as soon as we load.
```

Comment:
Use of /// comments is for creating documentation so that it's clear what this method or type does at the call site.

Your comment here is closer to an explanation of why the implementation behaves like it does.   

Consider using // instead here.